crypto/objects/obj_dat.c

```diff
@@ -668,9 +673,10 @@ int OBJ_create_objects(BIO *in)
                 } else
                     l = NULL;
             }
-        } else
+        } else {
             s = NULL;
-        if ((o == NULL) || (*o == '\0'))
+        }
+        if (s == NULL || *o == '\0')
```

Comment:
If we are deviating from original behaviour, then check for `s == NULL` is really sufficient...

Reply:
Today no, I keep it for a next PR ...